### PR TITLE
fix: support non-200 RPC error

### PR DESCRIPTION
**Pull request by 19Nazar**
## Pre-flight checklist

- [ ] I have read the [Contributing Guidelines on pull requests](https://github.com/near/near-api-js/blob/master/CONTRIBUTING.md).
- [ ] Commit messages follow the [conventional commits](https://www.conventionalcommits.org/) spec

## Motivation

I want to solve problems and contribute to Near ecosystems

## Related issues/PRs

Solved problam with support non-200 RPC error codes this issues [#1395](https://github.com/near/near-api-js/issues/1395)

### CODE REVIEW

**Comment by 19Nazar** [comment]
@race-of-sloths

**Comment by race-of-sloths** [comment]
@19Nazar Thank you for your contribution! Your pull request is now a part of the [Race of Sloths](https://race-of-sloths.com)!


[<picture>
    <source media="(prefers-color-scheme: dark)" srcset="https://race-of-sloths.com/images/bot-banner-dark.svg">
    <source media="(prefers-color-scheme: light)" srcset="https://race-of-sloths.com/images/bot-banner-light.svg">
    Shows inviting banner with latest news.
</picture>](https://race-of-sloths.com/latest-news)

[<picture>
    <source media="(prefers-color-scheme: dark)" srcset="https://badge.race-of-sloths.com/19Nazar?type=bot&pr=near/near-api-js/1404&theme=dark">
    <source media="(prefers-color-scheme: light)" srcset="https://badge.race-of-sloths.com/19Nazar?type=bot&pr=near/near-api-js/1404&theme=light">
    Shows profile picture for the author of the PR
</picture>](https://race-of-sloths.com/profile/19Nazar)


<details><summary>Current status: <i>waiting for scoring</i></summary>

We're waiting for maintainer to score this pull request with `@race-of-sloths score [0,1,2,3,5,8,13]` command. Alternatively, autoscoring [1,2] will be applied for this pull request

</details><details><summary>What is the Race of Sloths</summary>
<p>
Race of Sloths is a friendly competition where you can participate in challenges and compete with other open-source contributors within your normal workflow

For contributors:

- Tag @race-of-sloths inside your pull requests
- Wait for the maintainer to review and score your pull request
- Check out your position in the [Leaderboard](https://race-of-sloths.com/leaderboard)
- Keep weekly and monthly streaks to reach higher positions
- Boast your contributions with a dynamic picture of your [Profile](https://race-of-sloths.com/profile/19Nazar)

For maintainers:
- Score pull requests that participate in the Race of Sloths
- Engage contributors with fair scoring and fast responses so they keep their streaks
- Promote the Race to the point where the Race starts promoting you
- Grow the community of your contributors

Feel free to check [our website](https://race-of-sloths.com) for additional details!
</p>
</details> 

<details>
<summary>Bot commands</summary>

- For contributors
  - **Include a PR:** `@race-of-sloths include` to enter the Race with your PR
- For maintainers:
  - **Invite contributor** `@race-of-sloths invite` to invite the contributor to participate in a race or include it, if it's already a runner.
  - **Assign points:** `@race-of-sloths score [1/2/3/5/8/13]` to award points based on your assessment.
  - **Reject this PR:** `@race-of-sloths exclude` to send this PR back to the drawing board.
  - **Exclude repo:** `@race-of-sloths pause` to stop bot activity in this repo until `@race-of-sloths unpause` command is called

</details>

File: packages/providers/src/fetch_json.ts [hunk 1]
```diff
@@ -60,10 +60,12 @@ export async function fetchJsonRpc(url: string, json: JsonRpcRequest, headers: o
             throw new ProviderError(await res.text(), { cause: status });
```
Comment (PolyProgrammist) [inline packages/providers/src/fetch_json.ts:60]: So when status is other than `200`, the ok is false, so after we get `500`, `408` or another error, we stop here. So I suggest to move this line `if (!ok)` down the function

Also, have you tested your solution? If not, please test.

File: packages/providers/src/fetch_json.ts [hunk 2]
```diff
@@ -60,10 +60,12 @@ export async function fetchJsonRpc(url: string, json: JsonRpcRequest, headers: o
...
         }
 
-        if (status === 503) {
```
Comment (PolyProgrammist) [inline packages/providers/src/fetch_json.ts:63]: What is the reason for removing `503` error? It is still the case as I see in the `nearcore` code. It should be retried as well

**Comment by PolyProgrammist** [comment]
Hi @19Nazar please let me know if you are going to finish the PR

**Comment by 19Nazar** [comment]
> Hi @19Nazar please let me know if you are going to finish the PR

Hi, I'll get it done today

**Comment by 19Nazar** [comment]
@PolyProgrammist give feedback when you've checked it out

**Comment by 19Nazar** [comment]
@race-of-sloths

**Comment by frol** [comment]
@race-of-sloths score 5

**Pull request merged**